featuretools/computational_backends/pandas_backend.py

```diff
@@ -151,15 +170,48 @@ def calculate_all_features(self, instance_ids, time_last,
                 if not self.entityset.find_backward_path(start_entity_id=filter_eid,
                                                          goal_entity_id=eid):
                     entity_frames[eid] = eframes_by_filter[eid][eid]
+                    # precalculated features will only be placed in entity_frames,
+                    # and it's possible that that they are the only features computed
+                    # for an entity. In this case, the entity won't be present in
+                    # large_eframes_by_filter. The relevant lines that this case passes
+                    # through are 136-143
+                    if (large_eframes_by_filter is not None and
+                            eid in large_eframes_by_filter and eid in large_eframes_by_filter[eid]):
+                        large_entity_frames[eid] = large_eframes_by_filter[eid][eid]
 
             if filter_eid in self.feature_tree.ordered_feature_groups:
                 for group in self.feature_tree.ordered_feature_groups[filter_eid]:
                     if verbose:
-
                         pbar.set_postfix({'running': 0})
 
-                    handler = self._feature_type_handler(group[0])
-                    handler(group, entity_frames)
+                    test_feature = group[0]
+                    entity_id = test_feature.entity.id
+
+                    needs_all_values_type = \
+                        self.feature_tree.needs_all_values_differentiator(test_feature)
+
+                    input_frames = large_entity_frames
+                    if needs_all_values_type == "normal_no_dependent":
+                        input_frames = entity_frames
+
+                    handler = self._feature_type_handler(test_feature)
+                    result_frame = handler(group, input_frames)
+
+                    output_frames = []
+                    if needs_all_values_type in ['dependent', 'dependent_and_output']:
+                        # input is the full set of instances since
+                        # dependent feature needs all the values
+                        output_frames.append(large_entity_frames)
+                    if needs_all_values_type != "dependent":
+                        # input was selected (small) set of instances, or
+                        # feature itself is an output,
+                        # so need to place result
+                        # in selected (small) output frame
+                        output_frames.append(entity_frames)
+
+                    for frames in output_frames:
+                        index = frames[entity_id].index
+                        frames[entity_id] = result_frame.loc[index]
```

Comment:
you might want to use `result_frame.reindex()` here. you want to if some of the index values are going to be in result_frame.